engine/access/rest/handler.go

```diff
@@ -67,7 +71,7 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// create request decorator with parsed values
-	decoratedRequest := request.Decorate(r)
+	decoratedRequest := request.Decorate(r, h.chain)
```

Comment:
I'm afraid this is a sneaky but dirty way of solving this, it's nice because it doesn't require change of all handler functions but at the same time it's not great because chain ID has nothing to do with request object itself. For the sake of saving some time maybe we can leave it like that, but I would like to, once we have time, refactor this along with the refactor of handlers to be contained inside a struct that has chain, link generator, backend etc. This pattern is obviously starting to emerge

Reply:
i agree...there was no easy way to pass-in the chain ID other than completely changing the handler interface to accept additional arguments.
I have created this issue to take another stab at it https://github.com/dapperlabs/flow-go/issues/6132 and to pass chain id to CreateTransaction handler during server initialization.

Reply:
I feel we should have a conversation later about how we can improve some parts of the API, I'm sure you have bunch of ideas too and I want to share mine. So we can create some smaller PRs with refactors, but I would do that after we launch v1 and we are sure it's stable.

Reply:
my thoughts exactly! there is whole notion of session scoped, application scoped, request scope variables that is missing and that makes it difficult to change the implementation to factor in new pieces of information from different sources. But, like you said, I would really like to get this out first and have real world testing against it.